### ISSUE TITLE
Make sure database (and Django, if relevant) is initialised when accessing Sprout from the command line

### ISSUE THREAD

**Issue by martonvago** [seedcase-project/seedcase-sprout#502]
We want CLI commands to be accessible directly in the terminal (e.g. `> my-command --option 3`) when Sprout is installed, so we have to add our commands as entry points in `pyproject.toml`. This is a more complex setup than using custom admin commands (e.g. `> python manage.py my-command --option 3`).
When the app is accessed directly through a command, so not via `manage.py`, Django and the database have to be set up manually, including pointing to the settings file, running `django.setup()` and running the migrations.
Ideally (perhaps), these setup commands would be run once in a sort of post-install step, not before every command.

I’m not 100% sure what the best strategy would be for this, some possible options:
- Add a `setup.py` file with a post-install step
- Add a setup command that has to be run before all other commands
- Create a piece of middleware with the setup logic
- Put the setup logic in an initialisation file somewhere

Note that even if we can avoid having Django as a dependency of the core app, the database still needs to be initialised before we can run CLI commands that access it.

Related: https://github.com/seedcase-project/seedcase-sprout/issues/479

**Comment by martonvago** [comment]
If we don't have a database in `core` and don't depend on Django, then this is not needed. (See [this comment](https://github.com/seedcase-project/seedcase-sprout/issues/479#issuecomment-2225245118).)

**Comment by lwjohnst86** [comment]
Wonderful, thanks for this! I think we can keep this as a lower priority, and implement the actual CLI later. But at least make sure to design what ever we build with this in mind 😁

**Comment by lwjohnst86** [comment]
We decided to start the web app from scratch after the major design/architecture decisions and changes we made. No longer relevant.
